lib/client.js

```diff
@@ -286,6 +286,18 @@
         return true;
     };
 
+    Client.prototype.reauthenticate = function (auth) {
+
+        this.overrideReconnectionAuth(auth);
+
+        const request = {
+            type: 'reauth',
+            auth
+        };
+
+        return this._send(request, true).then(() => true);
```

Comment:
Either we don't care about server response, or we wait for actual response to resolve.